mlflow/keras.py

```diff
@@ -449,15 +451,18 @@ def __init__(self, keras_model, graph, sess):
         self._graph = graph
         self._sess = sess
 
-    def predict(self, dataframe):
+    def predict(self, data):
         # In TensorFlow < 2.0, we use a graph and session to predict
+        input_data = data
+        if isinstance(data, pd.DataFrame):
+            input_data = data.values
         if self._graph is not None:
             with self._graph.as_default():
                 with self._sess.as_default():
-                    predicted = pd.DataFrame(self.keras_model.predict(dataframe.values))
+                    predicted = pd.DataFrame(self.keras_model.predict(input_data))
```

Comment:
I think that we only want to return DataFrame if the input was DataFrame. If the was one of the numpy array based data types we should return the result as is.